src/lib.rs

```diff
@@ -71,6 +71,44 @@ impl PoseidonHasher {
     pub fn hash_no_pad(x: Vec<GoldilocksField>) -> Vec<u8> {
         PoseidonHash::hash_no_pad(&x).to_bytes()
     }
+
+    const EXPECTED_STORAGE_PREIMAGE_LEN: usize = 4 + 32 + 32 + 16;
+    
+    // This function should only be used to compute the quantus storage key for Transfer Proofs
+    // It breaks up the bytes input in a specific way that mimics how our zk-circuit does it
+    pub fn hash_storage(x: &[u8]) -> [u8; 32] {
```

Comment:
@dastansam is there a way to do this without asserts and unwraps? I know this is bad, but we need to use this in both StorageHasher on chain and in the api